### DISCUSSION TITLE
How do I Orama AI & Orama Search with Fumadocs?

### DISCUSSION THREAD

**Discussion by techwithanirudh** [fuma-nama/fumadocs#1039] [Q&A] [answered]
I’m trying to configure Orama AI and Orama Search within Fumadocs, but I need clarification on index types. Specifically:
- What type of index should I configure for Orama AI? (Rest API)
- What type of index is optimal for Orama Search? (Remote JSON)

Are these two seperate index? (I think so)

I'm checking this out

**Comment by fuma-nama** [comment]
For full-text search, you can follow the Orama Cloud integration guide which uses their Rest API source. And AI Search you can check their docs, I will recommend to use their official React library.

Fumadocs works well with Orama AI and inkeep.

**Reply by techwithanirudh** [reply]
@fuma-nama I’m running into an error while deploying a search index with Orama's REST API. The process gets to "Creating Orama database..." on the Orama dashboard, then stops and throws this error:

> "There was an error while deploying your index. Please check your dataset and schema, edit, and redeploy. If the error persists, please contact support at info@oramasearch.com."

### Here’s What I Did:
1. I created a new index, with the type of Rest API.
2. Added the necessary environment variables:
   ```plaintext
   ## Search
   ### Ingest
   ORAMA_SEARCH_INDEX_ID="index_id"

   ### UI
   NEXT_PUBLIC_ORAMA_SEARCH_API_KEY="not_here_yet"
   NEXT_PUBLIC_ORAMA_SEARCH_ENDPOINT="not_here_yet"
   ```
3. Included `sync-index.mts` from the Fumadocs site.
4. Added the `static.json` route.
5. Ran the sync script.
6. The dashboard then failed at "Creating Orama database..."

### Extra Info:

**Current Data and Schema Settings:**
- **Fields I Set as Searchable:**
    ```json
    {
      "id": "string",
      "tag": "string",
      "url": "string",
      "title": "string",
      "content": "string",
      "page_id": "string",
      "section": "string",
      "section_id": "string"
    }
    ```
- **static.json data:**
    ```json
    {
      "id": "/community-and-support",
      "structured": { "contents": [...], "headings": [...] },
      "tag": "community-and-support",
      "url": "/community-and-support",
      "title": "Community & Support",
      "description": "Join our Discord Community, or ping us on our socials."
    }
    ```

**Reply by fuma-nama** [reply, marked as answer]
Hmm, did you run `next build` and checked that the `static.json` output file is a non-empty JSON array? 
You can try redeploy, I noticed that Orama was down for a small period of time

**Reply by techwithanirudh** [reply]
@fuma-nama Yeah, static.json.body is present with all the data

Edit: i'll try redeploying

**Reply by techwithanirudh** [reply]
@fuma-nama Yeah, it was due to Orama’s downtime! It’s working now, thanks!
